server/src/main/java/org/elasticsearch/cluster/coordination/StartJoinRequest.java

```diff
@@ -16,34 +16,37 @@
 
 /**
  * Represents the action of requesting a join vote (see {@link Join}) from a node.
- * The source node represents the node that is asking for join votes.
+ *
+ * A node running for election as the new master in a new term will send requests
+ * to each node in the cluster to join the new cluster formed around itself as the
+ * master node.
```

Comment:
This is correct but not quite the full story. This message can also be used to trigger a new election for a candidate node other than the sender, which is how we implement graceful abdication: the current master broadcasts a `StartJoinRequest` with the abdication target as the `masterCandidateNode`.

Reply:
Thanks for the info! I reframed the comment, to be more abstract, and noted both use cases. It has more information about callers than I normally think appropriate, but for lack of a better place to document it 🤷‍♀️ 

I tried @link'ing abdicateTo, but couldn't because it's a private method. Linking seems great for keeping comments fresher, so that's a bummer. So it goes.